<?xml version="1.0"?>
<launch>
    <!-- Quad Name -->
    <arg name="quad_name" default=""/>
    <!-- "gazebo" or "arena" to indicate the environment-->
    <arg name="env" default=""/>

    <!-- MPC Parameters -->
    <arg name="t_mpc" default="1"/>
    <arg name="n_mpc" default="10"/>

    <!-- Use Ground truth Odometry instead of State Estimation -->
    <arg name="use_groundtruth" default="True"/>

    <!-- Parameter Indicating GPy  -->
    <arg name="with_gp" default="False"/>

    <!-- Parameter to indicate to compile acados model -->
    <arg name="compile" default="True"/>
    <param name="compile" value="$(arg compile)"/> 

    <group if="$(eval arg('env') == 'gazebo')" ns="gp_mpc">
        <rosparam file="$(find gp_rhce)/config/$(arg env)/gazebo_run.yaml"/>
        <rosparam file="$(find gp_rhce)/quads/$(arg quad_name)/$(arg quad_name)_mpc.yaml"/>
        <param name="environment" value="$(arg env)"/>
        <param name="quad_name" value="$(arg quad_name)"/>
        <param name="t_mpc" value="$(arg t_mpc)"/>
        <param name="n_mpc" value="$(arg n_mpc)"/>
        <param name="use_groundtruth" value="$(arg use_groundtruth)"/>
        <param name="with_gp" value="$(arg with_gp)"/>
        <!-- Node runs Acados Compiler code and GP_MPC node sequentially -->
        <node name="gp_mpc_wrapper" pkg="gp_rhce" type="sequential_mpc.py" output="screen"/>
    </group>        

    <group if="$(eval arg('env') == 'arena')" ns="gp_mpc">
        <rosparam file="$(find gp_rhce)/config/$(arg env)/arena_run.yaml"/>
        <rosparam file="$(find gp_rhce)/quads/$(arg quad_name)/$(arg quad_name)_mpc.yaml"/>
        <param name="environment" value="$(arg env)"/>
        <param name="quad_name" value="$(arg quad_name)"/>
        <param name="t_mpc" value="$(arg t_mpc)"/>
        <param name="n_mpc" value="$(arg n_mpc)"/>
        <param name="use_groundtruth" value="$(arg use_groundtruth)"/>
        <param name="with_gp" value="$(arg with_gp)"/>
        <!-- Node runs Acados Compiler code and GP_MPC node sequentially -->
        <node name="gp_mpc_wrapper" pkg="gp_rhce" type="sequential_mpc.py" output="screen"/>
    </group>
</launch>
